tests/Tenant/Feature/Filament/UserResource/RelationManagers/RolesRelationManagerTest.php

```diff
@@ -0,0 +1,146 @@
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use AdvisingApp\Authorization\Enums\LicenseType;
+use AdvisingApp\Authorization\Models\Role;
+use App\Filament\Resources\UserResource;
+use App\Filament\Resources\UserResource\Pages\EditUser;
+use App\Filament\Resources\UserResource\RelationManagers\RolesRelationManager;
+use App\Models\Authenticatable;
+use App\Models\User;
+use Filament\Forms\Components\Select;
+use Filament\Tables\Actions\AttachAction;
+
+use function Pest\Laravel\actingAs;
+use function Pest\Livewire\livewire;
+
+it('A non-super admin user cannot assign the super admin role.', function () {
+    $loggedInUser = User::factory()->licensed(LicenseType::cases())->create();
+
+    $user = User::factory()->create();
+
+    $loggedInUser->givePermissionTo(
+        'role.create',
+        'role.*.view',
+        'role.view-any',
+        'role.*.update',
+        'user.create',
+        'user.*.view',
+        'user.view-any',
+        'user.*.update',
+        'user.*.delete',
+        'user.*.restore',
+        'user.*.force-delete',
+    );
+
+    $loggedInUser->refresh();
+
+    actingAs($loggedInUser);
+
+    $superAdminRole = Role::query()->where('name', Authenticatable::SUPER_ADMIN_ROLE)->firstOrFail();
+
+    livewire(RolesRelationManager::class, [
+        'ownerRecord' => $user,
+        'pageClass' => EditUser::class,
+    ])
+        ->mountTableAction(AttachAction::class)
+        ->callTableAction(AttachAction::class, data: ['recordId' => $superAdminRole->getKey()])
+        ->assertHasTableActionErrors(['recordId' => 'The selected role is not allowed.']);
+});
+
+it('allows user which has sass global admin role to assign sass global admin role to other user', function () {
+    $user = User::factory()->create();
+    $user->assignRole(Authenticatable::SUPER_ADMIN_ROLE);
+
+    $second = User::factory()->create();
+
+    actingAs($user)
+        ->get(
+            UserResource::getUrl('edit', [
+                'record' => $second,
+            ])
+        )->assertSuccessful();
+
+    livewire(RolesRelationManager::class, [
+        'ownerRecord' => $second,
+        'pageClass' => EditUser::class,
+    ])
+        ->mountTableAction(AttachAction::class)
+        ->assertFormFieldExists('recordId', 'mountedTableActionForm', function (Select $select) {
+            $options = $select->getSearchResults(Authenticatable::SUPER_ADMIN_ROLE);
+
+            return ! empty($options) ? true : false;
+        })
+        ->assertSuccessful();
```

Comment:
Let's add an assertion after this to ensure that the user does now have the expected role.